### PR TITLE
Show visible features

### PR DESCRIPTION
**Pull request by domi4484**
image

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 9f3c17a42ad39202f03bea0d0793f85c2b2e5b83: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3213-a49450603fffeeb844b0dab0eb469ca25d7c2d38-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3213-a49450603fffeeb844b0dab0eb469ca25d7c2d38-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3213-a49450603fffeeb844b0dab0eb469ca25d7c2d38-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-3213-a49450603fffeeb844b0dab0eb469ca25d7c2d38-x86-android.apk)
</details>

File: src/core/multifeaturelistmodel.cpp [hunk 1]
```diff
@@ -47,6 +47,14 @@ void MultiFeatureListModel::setFeatures( QgsVectorLayer *vl )
   mSourceModel->setFeatures( requests );
 }
 
+void MultiFeatureListModel::setFeaturesForRectangle( QgsVectorLayer *vl, const QgsRectangle &rectangle )
```
Comment (suricactus) [inline src/core/multifeaturelistmodel.cpp:50]: I would name it Extent instead of Reactangle? No strong opinion.

**Review by suricactus — APPROVED** [review]
Lgtm, just the small comment.

File: src/core/multifeaturelistmodel.cpp (hunk 1 above)
Comment (domi4484) [reply src/core/multifeaturelistmodel.cpp]: @suricactus thanks i renamed it as you suggested

**Comment by nirvn** [comment]
@domi4484 , I'll have a look over the weekend. Nice feature. 

Two things come to mind when looking at the screenshot:
- we need an icon dedicated to this new feature, which I can take care of in a follow up PR
- these action buttons will eventually not scale up, 4 is probably the max, let's start thinking of alternatives

**Comment by suricactus** [comment]
Yup, thinking of big icon and tiny text below, we can fit a few per each row.

**Comment by domi4484** [comment]
@nirvn i made an icon inspired by QGIS attribute table but I am not satisfied. Maybe was the wrong inspiration...
image

**Comment by suricactus** [comment]
What if we add a "Filter" button on the top-right (or within the existing menu) on the feature list, that allows filtering only the currently visible features? In the future we can add more filters and sorting.

**Comment by nirvn** [comment]
@domi4484 , I pushed an icon here which I think works:
image

**Review by nirvn — APPROVED** [review]
Code looks good @domi4484 -- if you are happy with the icon, feel free to hit merge :)

**Comment by marioba** [comment]
@nirvn @suricactus I think it would be really nice to have like a funnel button in the title bar of the features list. So you can for example zoom on the map, filter to only visible features, then zoom again in the map, filter again and so on. I think this could be really useful.

**Pull request merged**